components/helm-broker/internal/broker/deprovision.go

```diff
@@ -113,8 +115,8 @@ func (svc *deprovisionService) doAsync(ctx context.Context, iID internal.Instanc
 func (svc *deprovisionService) do(ctx context.Context, iID internal.InstanceID, opID internal.OperationID, releaseName internal.ReleaseName) {
 
 	fDo := func() error {
-		if err := svc.helmDeleter.Delete(releaseName); err != nil {
-			return errors.Wrap(err, "while deleting helm release")
+		if err := svc.helmDeleter.Delete(releaseName); err != nil && !isErrReleaseNotFound(err, string(releaseName)) {
```

Comment:
imo to much in one line :D

please split it
```go
err := svc.helmDeleter.Delete(releaseName)
if err != nil && !isErrReleaseNotFound(err,releaseName) {
// ...
}
```

btw `isErrReleaseNotFound` it's your internal helper function, so more readable is to call it `sErrReleaseNotFound(err,releaseName)` instead of `isErrReleaseNotFound(err, string(releaseName))`. Haivng the string param makes sens when you export that to external lib but no when you have that here as helper func
what's more you already casting that in helper func :D 

`helmErrors.ErrReleaseNotFound(string(releaseName)).`